pkg/k8sutil/merge_test.go

```diff
@@ -113,9 +113,8 @@ func TestPodLabelsAnnotations(t *testing.T) {
 	for _, tc := range testCases {
 		result, err := MergePatchContainers(tc.base, tc.patches)
 		require.NoError(t, err)
-		if diff := pretty.Compare(result, tc.result); diff != "" {
-			t.Fatalf("Test %s: patch result did not match. diff: %s.", tc.name, diff)
-		}
+		diff := pretty.Compare(result, tc.result)
+		require.Empty(t, diff, "Test %s: patch result did not match. diff: %s.", tc.name, diff)
```

Comment:
Is this the same as an "equal" comparison? 🤔 

If it is, what do you think about just using a `require.Equal` instead of asserting the diff is empty?